### PR TITLE
Add Rust skeleton

### PR DESCRIPTION
**Pull request by MarSavar**
🦀

### CODE REVIEW

File: rust/script/setup [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+set -e
+
+if ! command -v rustc &> /dev/null
+then
+    echo "Could not find Rust. Installing..."
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
```
Comment (adamnfish) [inline rust/script/setup:8]: I'm not sure these scripts are supposed to have an impact on the machine outside of this project? I thought setup was supposed to do *project*-specific setup, like installing dependencies or downloading any project resources, as well as reset the project to its initial state. I think it's ok for a Rust project's setup to expect Rust?

This PR is fine either way, but perhaps we want to decide how much setup these repos should do.

I've looked at some of the other skeletons here and it looks like they are a mix of the both approaches. Some only do project setup and some try to set up the whole developer toolchain. Personally, I can imagine being quite frustrated if a project's setup script starts installing global compilers and language toolchains to my computer, but I know opinions will differ here!

For reference, this is the Github blogpost describing the `script/*` pattern, which does seem to suggest these should be project-level concerns?
https://github.blog/2015-06-30-scripts-to-rule-them-all/

Comment (MarSavar) [reply rust/script/setup:8]: Thanks for the feedback! 

I was a bit torn between whether or not I should include `rustup` in the setup script, and followed the example of other existing repos as you mention. 

I agree with you that that the `setup` script should really just be concerned with setting up the project. And the nice thing about Rust is that running `cargo new <project-name>` does all the heavy lifting and generates a project structure like the one in this PR, so there is no need for a `setup` script to begin with.

I think I'll just remove it altogether and add a friendly note about the toolchain in the README file.

**Review by michaelclapham — COMMENTED** [review]
Shall we add a start script? I know it'd probably just be ```cargo run``` but it's nice to provide when we can.

**Review by michaelclapham — APPROVED** [review]
Given you've addressed @adamnfish comments this looks good to me!

**Pull request merged**